### PR TITLE
[MIG][10.0] Migrate new module stock_picking_tracking to 10.0

### PR DESCRIPTION
**Pull request by tedsalmon**
This is a forward-port of https://github.com/OCA/stock-logistics-workflow/pull/239 to 10.0

cc @lasley

### CODE REVIEW

**Comment by lasley** [comment]
@tedsalmon please remove the v9 commit in place of this one instead - 7d9f2c9

Easiest way is checkout this branch, reset --hard to OCA/10.0, cherry pick 7d9f2c9, cherry pick c148d1bd402f94d08d1abd41bebb69a452b3ccf4

_Edit: Double check the 9.0 PR to make sure things haven't changed more too._

**Comment by lmignon** [comment]
@tedsalmon Do you plan to work on this PR and take into account the comments from @lasley?

**Comment by lasley** [comment]
This has been rebased now that dependency has been merged, and is ready for review

File: stock_picking_tracking/demo/stock_tracking_demo.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    @license LGPL-3 or later (http://www.gnu.org/licenses/lgpl.html).
+-->
+
+<odoo>
+
+    <record id="tracking_location_1" model="stock.picking.tracking.location">
+        <field name="city">Las Vegas</field>
+        <field name="state_id" ref="base.state_us_23" />
+        <field name="country_id" ref="base.us" />
+        <field name="zip_code">89123</field>
```
Comment (yvaucher) [inline stock_picking_tracking/demo/stock_tracking_demo.xml:14]: to rename `zip`

same for the 2 other tracking locations

File: stock_picking_tracking/models/stock_picking_tracking_event.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    state = fields.Selection(
+        OPERATION_STATES,
+        required=True,
+        default='unknown',
+        help="Status of the dispatch at time of event",
+    )
+    date_created = fields.Datetime(
+        string='Date',
+        readonly=True,
+        default=lambda s: fields.Datetime.now(),
```
Comment (yvaucher) [inline stock_picking_tracking/models/stock_picking_tracking_event.py:41]: Could directly be `default=fields.Datetime.now`

File: stock_picking_tracking/models/stock_picking_tracking_group.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    picking_id = fields.Many2one(
+        string='Stock Picking',
+        comodel_name='stock.picking',
+        required=True,
+    )
+    ref = fields.Char()
+    date_created = fields.Datetime(
+        readonly=True,
+        default=lambda s: fields.Datetime.now(),
```
Comment (yvaucher) [inline stock_picking_tracking/models/stock_picking_tracking_group.py:23]: `default=fields.Datetime.now`

**Review by yvaucher — CHANGES_REQUESTED** [review]
The demo data must be fixed to make the test pass.

Otherwise only few nitpicking.

**Comment by rousseldenis** [comment]
@tedsalmon

**Comment by tedsalmon** [comment]
@rousseldenis,

I'm no longer working with Odoo, nor do I have access to the repo where this PR is coming from, so I cannot help.

Thanks!
-Ted

File: stock_picking_tracking/README.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+=====
+
+Tracking information is located in the ``Tracking`` tab of Stock Pickings
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/154/10.0
+
+Known issues / Roadmap
```
Comment (Nikul-Chaudhary) [inline stock_picking_tracking/README.rst:22]: @tedsalmon If not content then remove section

File: stock_picking_tracking/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+{
+    "name": "Stock Picking Tracking",
+    "summary": "Adds a concept of event tracking for stock pickings",
+    "version": "10.0.1.0.0",
+    "category": "Inventory, Logistics, Warehousing",
+    "website": "https://laslabs.com/",
```
Comment (Nikul-Chaudhary) [inline stock_picking_tracking/__manifest__.py:10]: URL Should be `https://github.com/OCA/stock-logistics-workflow`

**Review by Nikul-Chaudhary — CHANGES_REQUESTED** [review]
Improve Code

**Comment by rousseldenis** [comment]
So, closing this.

**Pull request closed without merging**